src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/templates/objectGetUuidGeneratorService.php

```diff
@@ -0,0 +1,12 @@
+/**
+* @return \Spryker\Service\UtilUuidGenerator\UtilUuidGeneratorServiceInterface
+*/
+protected function getUuidGeneratorService()
+{
+    if ($this->_locator === null) {
+        $this->_locator = \Spryker\Zed\Kernel\Locator::getInstance();
+    }
+    $uuidGenerationService = $this->_locator->utilUuidGenerator()->service();
```

Comment:
I'm not sure it's a good idea to call locator and go through the whole architecture, generating uuid() itself is slow and now calling locator is making this process much slower than before! This will have huge effect on performance.

Reply:
@ehsanmx I don't know if this is really a performance problem or not, but we need to obey the architecture in order to be able to change stuff from project level.

Reply:
Loading service from locator is expensive call, so many glob() call in low level and I would say in infrastructure code we can avoid some slow parts

Reply:
and what if we do a static caching for the service itself?

`objectBaseAttribute.php`:
```
/**
 * @var UuidGeneratorServiceInterface
 */
private static $_uuidGeneratorService;
```
`objectGetUuidGeneratorService.php`:
```
protected function getUuidGeneratorService()
{
    if (static::$_uuidGeneratorService === null) {
        static::$_uuidGeneratorService = \Spryker\Zed\Kernel\Locator::getInstance()->utilUuidGenerator()->service();
    }

    return static::$_uuidGeneratorService;
}
```

Reply:
Yes this one is much better than before, then it's good from my side 👍, thank you @tamasnyulas !